GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1406,10 +1424,26 @@ private string GetNewProjectedShaForPlaceholder(string path)
                         FileData childFileData = childEntry as FileData;
                         string sha = childFileData.Sha.ToString();
 
-                        this.fileSystemVirtualizer.WritePlaceholderFile(childRelativePath, childFileData.Size, sha);
-                        updatedPlaceholderList.TryAdd(
-                            childRelativePath, 
-                            new PlaceholderListDatabase.PlaceholderData(childRelativePath, sha));
+                        FileSystemResult result = this.fileSystemVirtualizer.WritePlaceholderFile(childRelativePath, childFileData.Size, sha);
+                        switch (result.Result)
+                        {
+                            case FSResult.Ok:
+                                updatedPlaceholderList.TryAdd(
+                                    childRelativePath,
+                                    new PlaceholderListDatabase.PlaceholderData(childRelativePath, sha));
+                                break;
+
+                            case FSResult.FileOrPathNotFound:

```

Comment:
> Should we just check if the parent directory exists before creating the placeholder, to avoid this happening in the first place?

We could do that (I actually started with that approach), but I figured the more common case is that git has not deleted a folder as part of the projection change.

By using the current approach we avoid the cost of checking the disk for every directory that has to add a new file.

At this point I have not done any perf testing to measure the impact of checking if the directory exists, however, historically it's been worth it to avoid unnecessary I/O during projection changes.

I'm not sure that checking for the directory's existence first would make the code much simpler, but let me know if you disagree.